syntax = "proto3";

package NFMsg;

//events
enum EGameEventCode
{
	EGEC_SUCCESS								= 0;        //
	EGEC_UNKOWN_ERROR							= 1;		//
	EGEC_ACCOUNT_EXIST							= 2;        //
	EGEC_ACCOUNTPWD_INVALID						= 3;        //
	EGEC_ACCOUNT_USING							= 4;        //
	EGEC_ACCOUNT_LOCKED							= 5;        //
	EGEC_ACCOUNT_SUCCESS						= 6;        //
	EGEC_VERIFY_KEY_SUCCESS						= 7;        //
	EGEC_VERIFY_KEY_FAIL						= 8;        //
	EGEC_SELECTSERVER_SUCCESS					= 9;        //
	EGEC_SELECTSERVER_FAIL						= 10;       //
	

	EGEC_CHARACTER_EXIST						= 110;       //
	EGEC_SVRZONEID_INVALID						= 111;       //
	EGEC_CHARACTER_NUMOUT						= 112;       //
	EGEC_CHARACTER_INVALID						= 113;       //
	EGEC_CHARACTER_NOTEXIST						= 114;       //
	EGEC_CHARACTER_USING						= 115;       //
	EGEC_CHARACTER_LOCKED						= 116;       //
	EGEC_ZONE_OVERLOAD							= 117;       //
	EGEC_NOT_ONLINE								= 118;       //
	EGEC_ALREAY_IN_TEAM							= 119;       //
	EGEC_INVALID_TEAM_INVITE					= 120;       //
	EGEC_TEAM_FULL								= 121;       //
	EGEC_MAIL_UPDATE_ERROR						= 122;       //
	EGEC_MAIL_TONAME_INVALID					= 123;       //
	EGEC_MAIL_DELETE_ERROR						= 124;       //
	EGEC_MAIL_TOWNAME_EQUAL						= 125;       //
	EGEC_LACK_MONEY								= 126;       //
	EGEC_FULL_PACKAGE							= 127;       //
	EGEC_INVALID_ITEM							= 128;       //
	EGEC_INVALID_CONSIGNMENT					= 129;       //
	EGEC_NOT_ENOUGH_VIP							= 130;       //
	EGEC_NEED_LEVEL								= 131;       //
	EGEC_MAX_FRIEND_LIST						= 132;       //
	EGEC_MAX_ENEMY_LIST							= 133;       //
	EGEC_MAX_GANGS_LSIT							= 134;       //
	EGEC_LIMITS_TO_RIGHTS						= 135;       //
	EGEC_INVALID_GANGS_NAME						= 136;       //
	EGEC_REPEAT_ENTER_GANGS						= 137;       //
	EGEC_INVALID_GANGSMEMBER					= 138;       //
	EGEC_MASTER_LEAVE_GANGS						= 139;       //
	EGEC_GANGS_TITLE_OUTOF						= 140;       //
	EGEC_ENTER_GATEWAY_FAILD					= 141;       //
	EGEC_NO_SERVER4ZONE							= 142;       //
	EGEC_INVALID_SKILL							= 143;       //

	EGEC_CREATE_CLAN_SUCCESS					= 200;		 //
	EGEC_JOIN_CLAN_SUCCESS						= 201;		 //
	EGEC_LEAVE_CLAN_SUCCESS					= 202;		 //


}
enum ESynMsgID
{
	ESMI_UNKNOW						= 0;         //
}
enum EGameMsgID
{
	EGMI_UNKNOW						= 0;         //
	EGMI_EVENT_RESULT				= 1;         // for events
	EGMI_EVENT_TRANSPOND			= 2;         // for events
	EGMI_CLOSE_SOCKET				= 3;         // want to close some one

	EGMI_WTM_WORLD_REGISTERED		= 10;
	EGMI_WTM_WORLD_UNREGISTERED		= 11;
	EGMI_WTM_WORLD_REFRESH			= 12;

	EGMI_LTM_LOGIN_REGISTERED		= 20;
	EGMI_LTM_LOGIN_UNREGISTERED		= 21;
	EGMI_LTM_LOGIN_REFRESH			= 22;

	EGMI_PTWG_PROXY_REGISTERED		= 30;
	EGMI_PTWG_PROXY_UNREGISTERED	= 31;
	EGMI_PTWG_PROXY_REFRESH			= 32;

	EGMI_GTW_GAME_REGISTERED		= 40;
	EGMI_GTW_GAME_UNREGISTERED		= 41;
	EGMI_GTW_GAME_REFRESH			= 42;

	
	EGMI_DTW_DB_REGISTERED			= 60;
	EGMI_DTW_DB_UNREGISTERED		= 61;
	EGMI_DTW_DB_REFRESH				= 62;

	EGMI_STS_NET_INFO				= 70;
	
	EGEC_REQ_LAG_TEST				= 80;
	EGEC_ACK_GATE_LAG_TEST			= 81;
	EGEC_ACK_GAME_LAG_TEST			= 82;
	
	EGMI_STS_SERVER_REPORT          = 90;
	EGMI_STS_HEART_BEAT				= 100;
	//////////////////////////////////////////////////////////////////////////////////////
	EGMI_REQ_LOGIN								= 101;     	//
	EGMI_ACK_LOGIN								= 102;     	//
	EGMI_REQ_LOGOUT								= 103;		//

	EGMI_REQ_WORLD_LIST							= 110;			//
	EGMI_ACK_WORLD_LIST							= 111;			//
	EGMI_REQ_CONNECT_WORLD						= 112;			//
	EGMI_ACK_CONNECT_WORLD						= 113;
	EGMI_REQ_KICK_CLIENT_INWORLD				= 114;			//

	EGMI_REQ_CONNECT_KEY						= 120;         //
	EGMI_ACK_CONNECT_KEY						= 122;         //

	EGMI_REQ_SELECT_SERVER						= 130;			//
	EGMI_ACK_SELECT_SERVER						= 131;			//
	EGMI_REQ_ROLE_LIST							= 132;			//
	EGMI_ACK_ROLE_LIST							= 133;			//
	EGMI_REQ_CREATE_ROLE						= 134;			//
	EGMI_REQ_DELETE_ROLE						= 135;			//
	EGMI_REQ_RECOVER_ROLE						= 136;			//

	EGMI_REQ_LOAD_ROLE_DATA						= 140;			//
	EGMI_ACK_LOAD_ROLE_DATA						= 141;			//
	EGMI_REQ_SAVE_ROLE_DATA						= 142;			//
	EGMI_ACK_SAVE_ROLE_DATA						= 143;			//

	EGMI_REQ_ENTER_GAME							= 150;			//
	EGMI_ACK_ENTER_GAME							= 151;			//
	EGMI_REQ_LEAVE_GAME							= 152;			//
	EGMI_ACK_LEAVE_GAME							= 153;			//
	EGMI_REQ_SWAP_GAME							= 154;			//
	EGMI_REQ_SWAP_SCENE							= 155;			//
	EGMI_ACK_SWAP_SCENE							= 156;			//
	EGMI_REQ_SWAP_HOME_SCENE					= 157;			//
	EGMI_ACK_SWAP_HOME_SCENE					= 158;			//
	EGMI_REQ_ENTER_GAME_FINISH					= 159;			//
	EGMI_ACK_ENTER_GAME_FINISH					= 160;			//


	EGMI_ACK_OBJECT_ENTRY						= 200;			//
	EGMI_ACK_OBJECT_LEAVE						= 201;			//

	EGMI_ACK_OBJECT_PROPERTY_ENTRY				= 202;			//
	EGMI_ACK_OBJECT_RECORD_ENTRY				= 203;			//

	EGMI_ACK_PROPERTY_INT						= 210;			//
	EGMI_ACK_PROPERTY_FLOAT						= 211;			//
	EGMI_ACK_PROPERTY_STRING					= 212;			//
	//EGMI_ACK_PROPERTY_DOUBLE					= 213;			//
	EGMI_ACK_PROPERTY_OBJECT					= 214;			//
	EGMI_ACK_PROPERTY_VECTOR2                   = 215;
	EGMI_ACK_PROPERTY_VECTOR3                   = 216;
	EGMI_ACK_PROPERTY_CLEAR                   	= 217;

	EGMI_ACK_ADD_ROW							= 220;
	EGMI_ACK_REMOVE_ROW							= 221;
	EGMI_ACK_SWAP_ROW							= 222;
	EGMI_ACK_RECORD_INT							= 223;
	EGMI_ACK_RECORD_FLOAT						= 224;
	//EGMI_ACK_RECORD_DOUBLE					= 225;
	EGMI_ACK_RECORD_STRING						= 226;
	EGMI_ACK_RECORD_OBJECT						= 227;
	EGMI_ACK_RECORD_VECTOR2						= 228;
	EGMI_ACK_RECORD_VECTOR3						= 229;

	EGMI_ACK_RECORD_CLEAR						= 250;
	EGMI_ACK_RECORD_SORT						= 251;

	EGMI_ACK_DATA_FINISHED						= 260;

	EGMI_REQ_AI_ONWER							= 300;



	EGMI_REQ_SEARCH_OPPNENT                     = 1220;
	EGMI_ACK_SEARCH_OPPNENT                     = 1221;
	EGMI_REQ_END_OPPNENT						= 1222;
	EGMI_ACK_END_OPPNENT						= 1223;
	EGMI_REQ_CANCEL_SEARCH						= 1224;
	EGMI_ACK_CANCEL_SEARCH						= 1225;

	EGMI_REQ_MOVE								= 1230;
	EGMI_ACK_MOVE								= 1231;
	EGMI_REQ_MOVE_IMMUNE						= 1232;
	EGMI_ACK_MOVE_IMMUNE						= 1233;
	EGMI_REQ_STATE_SYNC							= 1234;
	EGMI_ACK_STATE_SYNC							= 1235;
	EGMI_REQ_POS_SYNC							= 1236;
	EGMI_ACK_POS_SYNC							= 1237;

	EGMI_REQ_SKILL_OBJECTX						= 1240;
	EGMI_ACK_SKILL_OBJECTX						= 1241;
	EGMI_REQ_SKILL_POS							= 1242;
	EGMI_ACK_SKILL_POS							= 1243;
	EGMI_REQ_ITEM_OBJECT						= 1244;
	EGMI_ACK_ITEM_OBJECT						= 1245;
	EGMI_REQ_ITEM_POS							= 1246;
	EGMI_ACK_ITEM_POS							= 1247;


	EGMI_REQ_CHAT								= 1250;
	EGMI_ACK_CHAT								= 1251;
	EGMI_REQ_SALE_ITEM							= 1252;
	EGMI_REQ_SPLIT_ITEM							= 1253;
	EGMI_REQ_PRODUCE_ITEM						= 1254;
	EGMI_REQ_PICK_ITEM							= 1255;
	EGMI_REQ_ACCEPT_TASK						= 1256;
	EGMI_REQ_COMPELETE_TASK						= 1257;



	EGMI_ACK_ONLINE_NOTIFY						= 1290; //
	EGMI_ACK_OFFLINE_NOTIFY						= 1291; //

	EGMI_REQ_CREATE_CLAN                       = 1300; //
	EGMI_ACK_CREATE_CLAN						= 1301; //
	EGMI_REQ_JOIN_CLAN                         = 1302; //
	EGMI_ACK_JOIN_CLAN							= 1303; //
	EGMI_REQ_LEAVE_CLAN                        = 1304; //
	EGMI_ACK_LEAVE_CLAN						= 1305; //
	EGMI_REQ_OPR_CLAN                        	= 1306; //
	EGMI_ACK_OPR_CLAN                        	= 1307; //
	EGMI_REQ_SEARCH_CLAN                       = 1308; //
	EGMI_ACK_SEARCH_CLAN                       = 1309; //
	EGMI_REQ_CANCEL_SEARCH_CLAN                = 1310; //


	EGEC_REQ_CREATE_CHATGROUP					= 1400;		 //
	EGEC_ACK_CREATE_CHATGROUP					= 1401;		 //
	EGEC_REQ_JOIN_CHATGROUP						= 1402;		 //
	EGEC_ACK_JOIN_CHATGROUP						= 1403;		 //
	EGEC_REQ_LEAVE_CHATGROUP					= 1404;		 //
	EGEC_ACK_LEAVE_CHATGROUP					= 1405;		 //
	EGEC_REQ_SUBSCRIPTION_CHATGROUP				= 1406;		 //
	EGEC_REQ_CANCELSUBSCRIPTION_CHATGROUP		= 1407;		 //
	EGEC_REQ_INVITE_CHATGROUP					= 1408;		 //
	EGEC_ACK_INVITE_CHATGROUP					= 1409;		 //
	EGEC_REQ_KICK_CHATGROUP						= 1410;		 //
	EGEC_ACK_KICK_CHATGROUP						= 1411;		 //

	EGEC_REQ_CHATGROUP_TO_TEAM					= 1450;		 //
	EGEC_ACK_CHATGROUP_TO_TEAM					= 1451;		 //

	EGEC_REQ_INTENSIFYLEVEL_TO_EQUIP			= 1500;		 //
	EGEC_ACK_INTENSIFYLEVEL_TO_EQUIP			= 1501;		 //

	EGEC_REQ_HOLE_TO_EQUIP						= 1502;		 //
	EGEC_ACK_HOLE_TO_EQUIP						= 1503;		 //

	EGEC_REQ_INLAYSTONE_TO_EQUIP				= 1504;		 //
	EGEC_ACK_INLAYSTONE_TO_EQUIP				= 1505;		 //

	EGEC_REQ_ELEMENTLEVEL_TO_EQUIP				= 1506;		 //
	EGEC_ACK_ELEMENTLEVEL_TO_EQUIP				= 1507;		 //


	EGEC_WEAR_EQUIP								= 1508;		 //
	EGEC_TAKEOFF_EQUIP							= 1509;		 //


	EGEC_REQ_SET_FIGHT_HERO						= 1520;		 //
	EGEC_REQ_SWITCH_FIGHT_HERO					= 1521;		 //
	EGEC_REQ_RELIVE_HERO						= 1522;		 //

    EGEC_REQ_SEND_MAIL                          = 1700; //
    EGEC_REQ_DRAW_MAIL                          = 1701; //
    EGEC_REQ_DELETE_MAIL                        = 1702; //

	//clone
	EGMI_ACK_CLONE_OBJECT_ENTRY						= 1800;			//
	EGMI_ACK_CLONE_OBJECT_LEAVE						= 1801;			//

	EGMI_ACK_CLONE_OBJECT_PROPERTY_ENTRY			= 1802;			//
	EGMI_ACK_CLONE_OBJECT_RECORD_ENTRY				= 1803;			//

	EGMI_ACK_CLONE_PROPERTY_INT						= 1810;			//
	EGMI_ACK_CLONE_PROPERTY_FLOAT					= 1811;			//
	EGMI_ACK_CLONE_PROPERTY_STRING					= 1812;			//
	//EGMI_ACK_CLONE_PROPERTY_DOUBLE				= 1813;			//
	EGMI_ACK_CLONE_PROPERTY_OBJECT					= 1814;			//

	//EGMI_ACK_CLONE_ADD_ROW						= 1820;
	//EGMI_ACK_CLONE_REMOVE_ROW						= 1821;
	//EGMI_ACK_CLONE_SWAP_ROW						= 1822;
	//EGMI_ACK_CLONE_RECORD_INT						= 1823;
	//EGMI_ACK_CLONE_RECORD_FLOAT					= 1824;
	//EGMI_ACK_CLONE_RECORD_DOUBLE					= 1825;
	//EGMI_ACK_CLONE_RECORD_STRING					= 1826;
	//EGMI_ACK_CLONE_RECORD_OBJECT					= 1827;
	//EGMI_ACK_CLONE_RECORD_CLEAR					= 1828;
	//EGMI_ACK_CLONE_RECORD_SORT					= 1829;

	EGMI_REQSWICHSERVER								= 1840;		//
	EGMI_ACKSWICHSERVER								= 1841;		//

	EGMI_REQ_CREATE_TEAM                       		= 1860; 	//
	EGMI_ACK_CREATE_TEAM							= 1861; 	//
	EGMI_REQ_JOIN_TEAM                         		= 1862; 	//
	EGMI_ACK_JOIN_TEAM								= 1863; 	//
	EGMI_REQ_LEAVE_TEAM                        		= 1864; 	//
	EGMI_ACK_LEAVE_TEAM								= 1865; 	//
	EGMI_REQ_INVITE_TEAM							= 1866; 	//
	EGMI_REQ_OPRMEMBER_TEAM							= 1867; 	//
	EGMI_ACK_OPRMEMBER_TEAM							= 1868; 	//

	EGMI_REQ_TEAM_ENTER_ECTYPE						= 1869; 	//
	EGMI_ACK_TEAM_ENTER_ECTYPE						= 1870; 	//


	/////////////////////MapModule/////////////////////////////////////
	EGMI_REQ_MAP_GRID_INFO							= 1900; 	//
	EGMI_ACK_MAP_GRID_INFO							= 1901; 	//
	EGMI_REQ_BIG_MAP_INFO							= 1902; 	//
	EGMI_ACK_BIG_MAP_INFO							= 1903; 	//

	EGMI_REQ_HOLD_MAP_GRID						= 1910; 	//
	EGMI_ACK_HOLD_MAP_GRID						= 1911; 	//
	EGMI_REQ_LEAVE_MSG_MAP_GRID					= 1912; 	//
	EGMI_ACK_LEAVE_MSG_MAP_GRID					= 1913; 	//
	EGMI_REQ_GET_MAP_GRID_AWARD					= 1914; 	//
	EGMI_ACK_GET_MAP_GRID_AWARD					= 1915; 	//

	EGMI_REQ_MAP_GRID_HUNTING					= 1916; 	//
	EGMI_ACK_MAP_GRID_HUNTING					= 1917; 	//
	EGMI_REQ_MAP_GRID_KING_WAR					= 1918; 	//
	EGMI_ACK_MAP_GRID_KING_WAR					= 1919; 	//

	//GAME SERVER TO WORLD SERVER
	EGMI_REQ_ADD_BUILDING						= 5000; 	//
	EGMI_REQ_ACK_BUILDING_LIST					= 5001; 	//
	

	EGMI_REQ_STORE_BUILDING_LIST						= 5010;

	/////////////////////FRIEND MSG START/////////////////////////////////////
	
	EGMI_REQ_ACK_FRIEND_LIST = 6000;
	EGMI_ACK_FRIEND_DELETE = 6001;
	EGMI_ACK_FRIEND_ADD = 6002;
	EGMI_ACK_INVITE_ADD = 6003;

	EGMI_REQ_ACK_SEND_INVITE = 6010;
	EGMI_REQ_ACK_ACCEPT_INVITE = 6011;
	EGMI_REQ_ACK_REJECT_INVITE = 6012;
	EGMI_REQ_ACK_IGNORE_INVITE = 6013;

	EGMI_REQ_ACK_BLOCK_PLAYER = 6014;
	EGMI_REQ_ACK_UNBLOCK_PLAYER = 6015;

	/////////////////////FRIEND MSG END/////////////////////////////////////
	
	/////////////////////TEAM MSG START/////////////////////////////////////
	
	EGMI_ACK_TEAM_INVITE = 6100;
	EGMI_ACK_TEAM_LIST = 6105;
	EGMI_ACK_TEAM_ADD = 6106;
	EGMI_ACK_TEAM_LEAVE = 6107;
	
	EGMI_REQ_ACK_SEND_TEAM_INVITE = 6120;
	EGMI_REQ_ACK_CANCEL_TEAM_INVITE = 6121;
	EGMI_REQ_ACK_ACCEPT_TEAM_INVITE = 6122;
	EGMI_REQ_ACK_REJECT_TEAM_INVITE = 6123;
	
	EGMI_REQ_ACK_SEND_TEAM_READY = 6130;
	EGMI_REQ_ACK_ACCEPT_TEAM_READY = 6131;
	EGMI_REQ_ACK_REJECT_TEAM_READY = 6132;
	
	/////////////////////TEAM MSG END/////////////////////////////////////


	/////////////////////GM MSG/////////////////////////////////////
	EGMI_REQ_CMD_PROPERTY_INT					= 10000;			//
	EGMI_REQ_CMD_PROPERTY_STR					= 10001;			//
	EGMI_REQ_CMD_PROPERTY_OBJECT				= 10002;			//
	EGMI_REQ_CMD_PROPERTY_FLOAT					= 10003;			//
	EGMI_REQ_CMD_RECORD_INT						= 10004;			//
	EGMI_REQ_CMD_RECORD_STR						= 10005;			//ַ
	EGMI_REQ_CMD_RECORD_OBJECT					= 10006;			//
	EGMI_REQ_CMD_RECORD_FLOAT					= 10007;			//
	EGMI_REQ_CMD_NORMAL							= 10008;			//


	EGMI_REQ_QUERY_SERVER_STATUS                 = 10200;

	/////////////////////Base SLG MSG/////////////////////////////////////
	EGMI_REQ_BUY_FORM_SHOP						= 20000;
	EGMI_ACK_BUY_FORM_SHOP						= 20001;
	EGMI_REQ_MOVE_BUILD_OBJECT					= 20002;
	EGMI_ACK_MOVE_BUILD_OBJECT					= 20003;

	EGMI_REQ_UP_BUILD_LVL						= 20101;
	EGMI_REQ_CREATE_ITEM						= 20102;
	EGMI_REQ_BUILD_OPERATE						= 20103;

	/////////////////////game-world MSG/////////////////////////////////////



	/////////////////////client-game MSG/////////////////////////////////////

}

///////////////////////////////////////////////////////////////////////////////////////////////////
//PropertyType
enum ESkillType
{
	EST_BRIEF_SINGLE 		= 0;//this kind of skill just can damage one object
	EST_BRIEF_GROUP 		= 1;//this kind of skill can damage multiple objects
	EST_BULLET_SINGLE		= 2;//this kind of bullet just can damage one object
	EST_BULLET_REBOUND		= 3;//this kind of bullet can damage multiple objects via rebound
	EST_BULLET_TARGET_BOMB	= 4;//this kind of bullet can damage multiple objects who around the target when the bullet touched the target object
	EST_BULLET_POS_BOMB		= 5;//this kind of bullet can damage multiple objects  who around the target when the bullet arrived the position
};

enum EItemType
{
    EIT_EQUIP   			= 0; //the equipment which can add props
	EIT_GEM   				= 1; //the gem ca be embed to the equipment
	EIT_SUPPLY   			= 2; //expendable items for player, such as a medicine that cures
	EIT_SCROLL   			= 3; //special items that can call a hero or others, special items can do what you want to do
}

//the sub items type of EIT_EQUIP
enum EGameEquipSubType
{
	EQUIPTYPE_WEAPON            = 0;    //
	EQUIPTYPE_HELMET            = 1;    //
	EQUIPTYPE_ARMOR             = 2;    //
	EQUIPTYPE_HAND              = 3;    //
	EQUIPTYPE_RING              = 4;    //
	EQUIPTYPE_BOOT              = 5;   	//
};

//the sub items type of EIT_GEM
enum EGameGemSubType
{
	EGGSTYPE_GEM_ATK           	= 0;    //
	EGGSTYPE_GEM_DEF           	= 1;    //

	EGGSTYPE_GEM_FIRE			= 3;    //
	EGGSTYPE_GEM_LIGHT			= 4;    //
	EGGSTYPE_GEM_WIND           = 5;    //
	EGGSTYPE_GEM_ICE          	= 6;    //
	EGGSTYPE_GEM_POISON         = 7;    //
	EGGSTYPE_GEM_DRAK         	= 8;    //

};

//the sub items type of EIT_ITEM
enum EGameSupplySubType
{
	EGIT_ITEM_WATER				= 0; 	//
	EGIT_ITEM_DIAMOND			= 1;	//
	EGIT_ITEM_CURRENCY			= 2; 	//
	EGIT_ITEM_EXP				= 3; 	//
	EGIT_ITEM_HP				= 4;	//
	EGIT_ITEM_MP				= 5;	//
	EGIT_ITEM_SP				= 6;	//
	EGIT_ITEM_PACK				= 7;	//
	EGIT_ITEM_MEMORY_POS		= 8;	//
};

//the sub items type of EIT_TOKEN
enum EGameScrollSubType
{
	EGTST_TOKEN_HERO_CARD				= 0; 	//
	EGTST_TOKEN_POSITION				= 1;	//
	EGTST_TOKEN_BOUNTY   				= 2; 	//
	EGTST_TOKEN_ACTIVITY				= 3;	//a token that can join a activity
	EGTST_TOKEN_BUILD					= 4;	//a token that can call a building or power
};


//
enum EGameItemExpiredType
{
    EGIET_NONE = 0;               	//
    EGIET_USE_TIMES = 1;            //
    EGIET_REAL_TIME = 2;            //
    EGIET_END_INDEX = 3;
};

/////////////////////////////////////////////////////////////////////////////////////////////////

enum EDrawDropItemState
{
    E_DRAW_STATE_NONE   = 0; //
    E_DRAW_STATE_GAIN   = 1; //
    E_DRAW_STATE_RECV   = 2; //
};
///////////////////////////////////////////////////////////////////////////////////////////////////
//
enum EGuildPowerType
{
    CLAN_POWER_TYPE_NORMAL   			= 0; //
	CLAN_POWER_TYPE_VICE_PRESIDENT		= 1; //
	CLAN_POWER_TYPE_PRESIDENT			= 2; //
	// TODO
}

//
enum ETaskState
{
    TASK_IN_PROCESS             = 0; //
    TASK_DONE                   = 1; //
    TASK_DRAW_AWARD             = 2; //
    TASK_FINISH                 = 3; //
}

//
enum ETaskType
{
    TASK_LEVEL_UPDATE_SOME_LEVEL        = 0; //
    TASK_PASS_ECTYPE                    = 1; //
    TASK_PASS_ECTYPE_COUNT              = 2; //
    TASK_PASS_SOME_TYPE_ECTYPE_COUNT    = 3; //
    TASK_UPDATE_SKILL_COUNT             = 4; //
    TASK_CONSUME_MONEY                  = 5; //
    TASK_CONSUME_DIAMOND                = 6; //
    TASK_ARENA_COUNT                    = 7; //
    TASK_KILL_MONSTER_COUNT             = 8; //
    TASK_KILL_SOME_MONSTER_COUNT        = 9; //
    // TODO
}
enum ESceneType
{
	SCENE_NORMAL = 0;//public town
	SCENE_SINGLE_CLONE = 1;//single boss underground city
	SCENE_MULTI_CLONE = 2;//world boss underground city
	SCENE_HOME = 3;//private
	SCENE_CLAN = 4;//private
}
//
enum EPVPType
{
    PVP_HOME = 0;
    PVP_INDIVIDUAL = 1;
    PVP_DOTA = 2;
    PVP_SURVIVAL = 3;
    PVP_CLAN_WAR = 4;
    PVP_CLAN_DUNGEON = 5;
}

enum EShopType
{
	EST_UNKNOW		= 0;//
	EST_BUILDING 	= 1;//
	EST_GOLD		= 2;//
	EST_ITEM		= 3;//
	EST_EQUIP		= 5;//
	EST_GEM			= 6;//
	EST_CARD		= 7;//
}


///////////////////////////////////////////////////////////////////////////////////////////////////

enum ERankType
{
	ERT_UNKNOW					= 0;
	ERT_PLAYER_LEVEL 			= 1;
	ERT_FIGHT_VALUE				= 2;
	ERT_PLAYER_MONEY			= 3;
	ERT_CLAN_LEVEL				= 4;
	ERT_CLAN_KINGSOURCE		= 5;
}

///////////////////////////////////////////////////////////////////////////////////////////////////

//NPCTypenpc
enum ENPCType
{
    ENPCTYPE_NORMAL	= 0;			  //
    ENPCTYPE_HERO 	= 1;              //
    ENPCTYPE_TURRET = 2;              //

};
enum ESubNPCType_TURRET
{
    ENPCTYPE_SUB_STORAGE 		= 0;
    ENPCTYPE_SUB_WEAPONE 		= 1;
    ENPCTYPE_SUB_GUARD 			= 2;
    ENPCTYPE_SUB_BOMB 			= 3;
    ENPCTYPE_SUB_TRAP 			= 4;

};
enum EMeleeType
{
    ETYPE_MELEE	= 0;			  //
    ETYPE_MELEE_DASH 	= 1;              //
    ETYPE_RANGER = 2;              //

};

enum EEffectObjType
{
    ETARGETTYPE_ENEMY	= 0;			  //
    ETARGETTYPE_SELF 	= 1;              //
    ETARGETTYPE_FRIEND  = 2;              //

};

//PropertyType
enum EPropertyType
{
	EPT_UNKNOW				= 0;//
	EPT_STRENGTH 			= 1;//
	EPT_AGILITY				= 2;//
	EPT_INTELLIGENCE		= 3;//
};

//
enum EGameElementType
{
	EELEMENTTYPE_FIRE				= 0;    //
	EELEMENTTYPE_LIGHT				= 1;    //
	EELEMENTTYPE_WIND              	= 2;    //
	EELEMENTTYPE_ICE          		= 3;    //
	EELEMENTTYPE_POISON          	= 4;    //
	EELEMENTTYPE_DARK          		= 5;    //
};

enum EBattleType
{
	EBT_SINGLE_MODE = 0;
	EBT_MULTI_MODE = 1;
	EBT_CLAN_MODE = 2;
	EBT_CLAN_BOSS_MODE = 3;
	EBT_WORLD_BOSS_MODE = 4;
	EBT_MOBA_MODE = 5;
	EBT_SURVIVAL_MODE = 6;
};
